packages/SwingSet/docs/vat-upgrade.md

```diff
@@ -94,32 +100,33 @@ To enable upgrade, the v1 code must stash important data in durable storage, and
 
 For upgrade-important in-RAM objects, the v1 code needs to store a "reattachment handle" (name and API TBD) in durable storage. This gives the v2 code the authority to take over the identity of these exported objects.
 
-For upgrade-important virtual objects (specifically durable virtual objects), the v1 code must store the "durable kind handle" in durable storage. This handle is created before calling `makeDurableKind`, like this:
+For upgrade-important durable objects, the v1 code must store the "durable kind handle" in durable storage. This is done automatically by `prepareExoClass()` and `prepareExoClassKit()`, but can also be done manually (see [Virtual and Durable Objects](./virtual-objects.md#virtual-and-durable-objects)).
 
 ```js
-const fooHandle = allocateDurableKindHandle();
-const makeFoo = makeDurableKind(fooHandle, fooBehavior);
-someDurableStorage.set('foo handle', fooHandle);
+import { M } from '@agoric/store';
+const FooI = M.interface('foo', fooMethodGuards);
+// someDurableMap should generally be reachable from baggage.
+const makeFoo = prepareExoClass(someDurableMap, 'foo', fooI, initFoo, fooMethods);
 ```
 
 The v1 code can also store imported objects (Presences) and plain data in a durable collection. Durable collections are themselves durable objects, so they can be nested:
 
 ```js
-const childMap = makeScalarDurableMapStore();
-const parentMap = makeScalarDurableMapStore();
-parentMap.init('child', childMap);
+const parentMap = makeScalarBigMapStore(parentLabel, { durable: true });
 baggage.init('parent', parentMap);
+const childMap = makeScalarBigMapStore(childLabel, { durable: true });
+parentMap.init('child', childMap);
 ```
 
-The 'baggage' is a special instance of `makeScalarDurableMapStore()`. The backing data is stored in a well-known per-vat location, so each version can be given a reference. These are, of course, distinct JavaScript `Object`s, just like running the same program twice results in distinct objects within each invocation. But for every piece of data that v1 wrote into the baggage, v2 can read an equivalent item from the baggage it receives.
+The 'baggage' is a special instance of `makeScalarBigMapStore`. The backing data is stored in a well-known per-vat location, so each version can be given a reference. These are, of course, distinct objects, just like running the same program twice results in distinct objects within each invocation. But for every piece of data that v1 wrote into the baggage, v2 can read an equivalent item from the baggage it receives.
```

Comment:
Maybe reconsider removing the "JavaScript `Object`" discriminator: by saying merely "distinct objects", we're being casual about the sort of abstract "object stored in baggage" or "object with an external identity", vs the very specific "of course it must be a different `Object` because it's a different program"

From one point in the abstraction stack, the `baggage` in v1 and the `baggage` in v2 *are* the same object: they have equivalent contents, they have the same external identity (vref). But reminding the reader that they're different `Object`s reminds them that v1 and v2 are different programs, so they shouldn't expect anything outside of `baggage` to carry over.

Reply:
I've refactored this paragraph a bit more, please take another look.